cli/cmd/compliance.go

```diff
@@ -263,3 +283,62 @@ func buildComplianceReportTable(detailsTable, summaryTable, recommendationsTable
 	}
 	return mainReport.String()
 }
+
+func filterRecommendations(recommendations []api.ComplianceRecommendation) ([]api.ComplianceRecommendation, string) {
+	var filtered []api.ComplianceRecommendation
+	for _, r := range recommendations {
+		if matchRecommendationsFilters(r) {
+			filtered = append(filtered, r)
+		}
+	}
+	if len(filtered) == 0 {
+		return filtered, "There are no recommendations with the specified filters.\n"
+	}
+
+	cli.Log.Debugw("filtered recommendations", "recommendations", filtered)
+	return filtered, fmt.Sprintf("%v of %v recommendations showing \n", len(filtered), len(recommendations))
+}
+
+func matchRecommendationsFilters(r api.ComplianceRecommendation) bool {
+	var results []bool
+
+	// severity returns specified threshold and above
+	if compCmdState.Severity != "" {
+		sevThreshold, _ := eventSeverityToProperTypes(compCmdState.Severity)
+		results = append(results, r.Severity <= sevThreshold)
+	}
+
+	if compCmdState.Category != "" {
+		category := strings.ReplaceAll(compCmdState.Category, "-", " ")
+		results = append(results, strings.EqualFold(r.Category, category))
+	}
+
+	if compCmdState.Service != "" {
+		results = append(results, strings.EqualFold(r.Service, compCmdState.Service))
+	}
+
+	if compCmdState.Status != "" {
+		results = append(results, r.Status == statusToProperTypes(compCmdState.Status))
+	}
+
+	return !array.ContainsBool(results, false)
+}
+
+func filtersEnabled() bool {
+	return compCmdState.Category != "" || compCmdState.Status != "" || compCmdState.Severity != "" || compCmdState.Service != ""
+}
+
+func statusToProperTypes(status string) string {
```

Comment:
Could we add some test that checks these two functions, I see we are only testing a single path:
<img width="1294" alt="Screen Shot 2021-03-02 at 7 15 27 AM" src="https://user-images.githubusercontent.com/5712253/109654001-e9ff7380-7b1e-11eb-9f9c-b2daf834e5df.png">

Since this is coming from `complianceReportRecommendationsTable()` we might need to do integration tests but not sure about it. 🤔